idx/cassandra/cassandra.go

```diff
@@ -225,8 +232,12 @@ func (c *CasIdx) Add(data *schema.MetricData, partition int32) error {
 				log.Debug("cassandra-idx def hasnt been seem for a while, updating index.")
 				def := schema.MetricDefinitionFromMetricData(data)
 				def.Partition = partition
-				c.MemoryIdx.AddDef(def)
-				c.writeQueue <- writeReq{recvTime: time.Now(), def: def}
+				err := c.MemoryIdx.AddOrUpdateDef(def)
+				if err != nil {
+					c.writeQueue <- writeReq{recvTime: time.Now(), def: def}
+				}
+				statUpdateDuration.Value(time.Since(pre))
+				return err
```

Comment:
at this point `err` can not be anything else than `nil`, the first line after this `if` block is doing `return nil`, so you might as well just skip this line.